src/qcodes/dataset/descriptions/dependencies.py

```diff
@@ -407,6 +407,64 @@ def _from_graph(cls, graph: nx.DiGraph[str]) -> InterDependencies_:
         new_interdependencies._graph = graph
         return new_interdependencies
 
+    def collect_all_related_parameters(
+        self, initial_params: set[ParamSpecBase]
+    ) -> set[ParamSpecBase]:
```

Comment:
Given that Interdependencies_ is now implemented as a networkx graph it should be possible to simplify this significantly